lib/plugins/stats.js

```diff
@@ -34,24 +48,82 @@ module.exports = async robot => {
     return github.paginate(req, res => res.data)
   }
 
-  async function popularInstallations (installations) {
-    let popular = await Promise.all(installations.map(async installation => {
-      const github = await robot.auth(installation.id)
-
-      const req = github.apps.getInstallationRepositories({per_page: 100})
-      const repositories = await github.paginate(req, res => {
-        return res.data.repositories.filter(repository => !repository.private)
+/**
+  * Build a search query that includes all repos the app is installed on.
+  *
+  * @returns {Array} - Array of Query Parameters
+  *
+  * @example
+  * [
+  *   Query {
+  *     id: '12345' // An installation id used to obtain an auth token later
+  *     queryString: 'org:github' // Search string scoped to the user/org
+  *   }
+  * ]
+  */
+  function buildQuery (installations) {
+    let queryParams = []
+    installations.map(installation => {
+      queryParams.push({
+        search: 'org:' + installation.account.login + ' ',
+        id: installation.id
       })
-      const account = installation.account
+    })
+    return queryParams
+  }
+
+  /**
+   * Returns the top 10 repos (ordered by stars) from the list of repos
+   * constructed from buildQuery()
+   * @returns {Array} - Array of all GraphQL nodes
+   *
+   * @example
+   *    [ {
+   *        name: 'react', stargazers: { totalCount:	85592 }, owner: { login:	'facebook' }
+   *      },
+   *      {
+   *        name: 'gitignore', stargazers: { totalCount:	60035 }, owner: { login:	'github' }
+   *      } ]
+   */
+  async function getPopularGraphQL (installations) {
+    let Query = buildQuery(installations)
+    const queryString = Query.reduce((a, b) => {
+      return a.search + b.search
+    })
 
-      account.stars = repositories.reduce((stars, repository) => {
-        return stars + repository.stargazers_count
-      }, 0)
+    const github = await robot.auth(Query[0].id)
+    const token = github.auth.token
```

Comment:
@gr2m @bkeepers This doesn't seem to be working anymore after syncing with `master`:

```
TypeError: Cannot read property 'token' of undefined
```

The GitHub object that's returned doesn't have the auth object like it did earlier. I couldn't find anything in the changelog in probot, but maybe we just aren't exposing that anymore since we're explicitly pulling in `octokit/rest.js`? Any thoughts on how to pull in the token?

Reply:
Doh, that's because of the refactoring in octokit/rest.js. I ran into that in a project as well.

We're going to have to extract [this code](https://github.com/probot/probot/blob/9394ca1db32b72bdd17a5cd9dd606a7d4eec0f00/lib/robot.js#L148-L153), which creates an installation token, into a separate method. For now feel free to define `robot.getToken`.